### DISCUSSION TITLE
Day 9 of 40

### DISCUSSION THREAD

**Discussion by WeenAIthDev** [WeenAIthDev/X-like-premium-posts-on-GitHub#17] [General]
Okay so yesterday I was working on the n8n pipeline still a WIP :)

Understood how to analyse features thru a histogram and how to deal with capped values especially if they are effecting your target column learn abt stratas (dividing a feature into various categories for better predicts by being a better representation for your data) , learnt abt various ways to split data into train and test split and why the train and test split function from sklearn.model_selection is preferred over a customs functions like hashing a certain feature combination value and determine if the hash is 20% of total hash or setting a random seed value of 42 (choosen cause it answers question of life and if we see God did complete creating all his creations as a project life v1 somewhere between the end of 6th day and 7th day so it makes sense as in Genesis) just before shuffling the set to choose the index till where the test set should end then saw how much of your data should be saved as test set 20% but for not choosing a random seed and choosing one this custom func would fail if we updated the dataset or kept choosing the test set over again as the model could see the whole dataset which is not the intention , then if we take the hash function of the index of the dataset then we have to ensure that the test values are added at the end and the values don't get erased , for this specific dataset a good custom function was to choose the hash of a latitude and longitude combo as those will be fixed for many years and how train-test split function is more advantageous over all the custom functions as it has all the attributes like random value, stratify, we can add data with same size it works on the split shuffle mechanism we just discussed abt from the Hands on ML book

And looked thru past commits of the repo I'm currently trying to resolve issues

Stay tuned for more .....
